File: src/main/java/com/jss/osiris/modules/accounting/model/AccountingBalanceSearch.java
package com.jss.osiris.modules.accounting.model;

import java.time.LocalDateTime;
import java.util.List;

public class AccountingBalanceSearch {
  private AccountingAccountClass accountingClass;
  private AccountingAccount accountingAccount;
  private List<PrincipalAccountingAccount> principalAccountingAccounts;

  private LocalDateTime startDate;
  private LocalDateTime endDate;

  private Boolean isFromAs400;

  public Boolean getIsFromAs400() {
    return isFromAs400;
  }

  public void setIsFromAs400(Boolean isFromAs400) {
    this.isFromAs400 = isFromAs400;
  }

  public AccountingAccountClass getAccountingClass() {
    return accountingClass;
  }

  public void setAccountingClass(AccountingAccountClass accountingClass) {
    this.accountingClass = accountingClass;
  }

  public AccountingAccount getAccountingAccount() {
    return accountingAccount;
  }

  public void setAccountingAccount(AccountingAccount accountingAccount) {
    this.accountingAccount = accountingAccount;
  }

  public LocalDateTime getStartDate() {
    return startDate;
  }

  public void setStartDate(LocalDateTime startDate) {
    this.startDate = startDate;
  }

  public LocalDateTime getEndDate() {
    return endDate;
  }

  public void setEndDate(LocalDateTime endDate) {
    this.endDate = endDate;
  }

  public List<PrincipalAccountingAccount> getPrincipalAccountingAccounts() {
    return principalAccountingAccounts;
  }

  public void setPrincipalAccountingAccounts(List<PrincipalAccountingAccount> principalAccountingAccounts) {
    this.principalAccountingAccounts = principalAccountingAccounts;
  }

}

File: client/src/app/modules/accounting/model/AccountingBalanceSearch.ts
import { AccountingAccount } from "./AccountingAccount";
import { AccountingAccountClass } from "./AccountingAccountClass";
import { AccountingJournal } from "./AccountingJournal";
import { PrincipalAccountingAccount } from './PrincipalAccountingAccount';

export interface AccountingBalanceSearch {
  accountingClass: AccountingAccountClass;
  accountingAccount: AccountingAccount | undefined;
  accountingJournal: AccountingJournal | undefined;
  principalAccountingAccounts: PrincipalAccountingAccount[] | undefined;
  startDate: Date | undefined;
  endDate: Date | undefined;
  isFromAs400: boolean;
  doNotDisplayZeroTiersAccounts: boolean;
}
